tuist/tuist
# Improve warning experience

### Short description 📝
Warnings were scattered throughout the logs. This is not an ideal experience because when logs are dense, the warnings might go unnoticed. This PR improves that by adding a global utility that can hold warnings output during the execution of Tuist, and flush all of them out to the standard output right before the process exits.

### How to test the changes locally 🧐
If you run `ARGS="--no-open" make tuist/generate"` you should see some warnings.

<img width="1278" alt="image" src="https://github.com/tuist/tuist/assets/663605/ae642b32-33b3-4b54-9933-b14529d2cd3f">

### Contributor checklist ✅


### Reviewer checklist ✅
